manager/sinks.go

```diff
@@ -22,9 +22,9 @@ import (
 	sink_api "github.com/GoogleCloudPlatform/heapster/sinks/api/v1"
 )
 
-func newSinks() ([]sink_api.ExternalSink, error) {
+func newSinks(sinkUris []string) ([]sink_api.ExternalSink, error) {
```

Comment:
How we do this is so nasty... (not your code). This whole overloading of a URI and using a string instead of a strongly-typed thing really doesn't sit well with me. WDYT of making this a slightly stronger and friendlier API? At the very least we can do:

```Go
type SinkConfig struct {
    // Name of the sink.
    Name string
    // Arguments for configuring the sink.
    ConfigArgs string
}
```

It won't be as nice for flags, but it'll work much better for a REST API. I'd even argue to make the SinkArgs more concrete. Maybe make this a struct with configs for all the sinks?

```Go
type SinkConfig struct {
    GCM GCMConfig `json:"gcm,omitempty"`
    InfluxDB InfluxDBConfig `json:"gcm,omitempty"`
    ...
}
```